sdk/python/lib/pulumi/provider/experimental/analyzer.py

```diff
@@ -491,78 +447,49 @@ def analyze_property(
                 ref=ref,
                 optional=optional,
                 plain=plain,
-                description=self.get_docstring(typ.__name__, name),
+                description=self.get_docstring(typ, name),
             )
         else:
             raise ValueError(f"Unsupported type '{arg}' for '{typ.__name__}.{name}'")
 
-    def find_docstrings(self, path: Path) -> dict[str, dict[str, str]]:
-        """
-        find_docstrings returns the docstrings for all the attributes of all
-        the classes in `self.path.
-
-        Unfortunately, only class docstrings are available at runtime, the
-        docstrings of the attributes are not available. Instead of relying on
-        runtime information we parse the source code to extract the docstrings.
-        """
-        docs: dict[str, dict[str, str]] = {}
-        for file_path in self.iter(path):
-            if file_path.suffix != ".py":
-                continue
-            with open(file_path) as f:
-                src = f.read()
-                try:
-                    t = ast.parse(src)
-                    docs.update(self.find_docstrings_in_module(t))
-                except SyntaxError as e:
-                    rel_path = file_path.relative_to(path)
-                    raise Exception(f"Failed to parse {rel_path}: {e}") from e
-        return docs
-
-    def find_docstrings_in_module(self, mod: ast.Module) -> dict[str, dict[str, str]]:
-        """
-        Find the docstrings for all the class attributes in the module. The
-        return dict has an entry for each class name, with a dict of attribute
-        name to docstring.
-
-        We look for all the (top level) class definitions in the module, and for
-        each class we look for attribute assignments with type annotations. If
-        we find such an assignment, we look for a string constant right after
-        the assignment, which we assume is the docstring for the attribute.
```

Comment:
Not sure if there's anything valuable here/that we should docstring for the remaining `get_docstring` method?

Reply:
Yes, I'll re-add a docstring here.